src/background/providers/hcaptcha.ts

```diff
@@ -24,19 +81,179 @@ export class HcaptchaProvider implements Provider {
         this.callbacks.updateIcon(this.getBadgeText());
     }
 
+    private handleUrl(url: URL) {
+        const reqUrl = url.origin + url.pathname;
+        const isIssuerUrl = !!reqUrl.match(ISSUER_REGEX);
+
+        let isSpendUrl;
+        if (HCAPTCHA_URLS.some((url: string) => reqUrl === url)) {
+            isSpendUrl = false;
+        } else {
+            isSpendUrl = !!reqUrl.match(SPEND_REGEX);
+        }
+
+        return {
+            reqUrl,
+            isIssuerUrl,
+            isSpendUrl,
+        };
+    }
+
+    private async getCommitment(version: string): Promise<{ G: string; H: string }> {
+        const keyPrefix = 'commitment-';
+        const cached = this.storage.getItem(`${keyPrefix}${version}`);
```

Comment:
consider declaring
```
const key = `${keyPrefix}${version}`
``` as you use it multiple time

Reply:
done!